protocol/synthetix/contracts/storage/CollateralConfiguration.sol

```diff
@@ -215,17 +220,33 @@ library CollateralConfiguration {
      * E.g: USDC uses 6 decimals of precision, so this would upscale it by 12 decimals.
      * @param self The CollateralConfiguration object corresponding to the collateral type being converted.
      * @param tokenAmount The token amount, denominated in its native decimal precision.
-     * @return The converted amount, denominated in the system's 18 decimal precision.
+     * @return amountD18 The converted amount, denominated in the system's 18 decimal precision.
      */
     function convertTokenToSystemAmount(
         Data storage self,
         uint256 tokenAmount
-    ) internal view returns (uint256) {
+    ) internal view returns (uint256 amountD18) {
         // this extra condition is to prevent potentially malicious untrusted code from being executed on the next statement
         if (self.tokenAddress == address(0)) {
             revert CollateralNotFound();
         }
 
-        return (tokenAmount * DecimalMath.UNIT) / (10 ** IERC20(self.tokenAddress).decimals());
+        try IERC20(self.tokenAddress).decimals() returns (uint8 decimals) {
+            /// @dev if decimals() is not defined, but an existing fallback function returns a
+            /// bytes representation of uint8, this logic may have unexpected results
+            if (decimals == 0) {
+                amountD18 = tokenAmount;
+            } else if (decimals <= 18) {
```

Comment:
technially this is not the most gas efficient if `decimals == 18`. might be best to just see if `decimals == 18` at the beginning of the function and return early if so

Reply:
good idea! early exit is always better